internal/auth/repository/eventsourcing/handler/user_session.go

```diff
@@ -254,158 +221,98 @@ func (u *UserSession) Reduce(event eventstore.Event) (_ *handler.Statement, err
 				}
 			}
 			return u.updateSession(session, event)
-		case user.UserV1PasswordChangedType,
-			user.UserV1MFAOTPRemovedType,
-			user.UserV1ProfileChangedType,
-			user.UserLockedType,
-			user.UserDeactivatedType,
-			user.HumanPasswordChangedType,
-			user.HumanMFAOTPRemovedType,
-			user.HumanProfileChangedType,
-			user.HumanAvatarAddedType,
-			user.HumanAvatarRemovedType,
-			user.UserDomainClaimedType,
-			user.UserUserNameChangedType,
-			user.UserIDPLinkRemovedType,
-			user.UserIDPLinkCascadeRemovedType,
-			user.HumanPasswordlessTokenRemovedType,
-			user.HumanU2FTokenRemovedType:
-			sessions, err := u.view.UserSessionsByUserID(event.Aggregate().ID, event.Aggregate().InstanceID)
-			if err != nil || len(sessions) == 0 {
-				return err
-			}
-			if err = u.appendEventOnSessions(sessions, event); err != nil {
-				return err
-			}
-			if err = u.view.PutUserSessions(sessions); err != nil {
-				return err
-			}
-			return nil
-		case org.OrgDomainPrimarySetEventType:
-			return u.fillLoginNamesOnOrgUsers(event)
-		case user.UserRemovedType:
+		}), nil
+	case user.UserLockedType,
+		user.UserDeactivatedType:
+		return handler.NewUpdateStatement(event,
+			[]handler.Column{
+				handler.NewCol("passwordless_verification", time.Time{}),
+				handler.NewCol("password_verification", time.Time{}),
+				handler.NewCol("second_factor_verification", time.Time{}),
+				handler.NewCol("second_factor_verification_type", domain.MFALevelNotSetUp),
+				handler.NewCol("multi_factor_verification", time.Time{}),
+				handler.NewCol("multi_factor_verification_type", domain.MFALevelNotSetUp),
+				handler.NewCol("external_login_verification", time.Time{}),
+				handler.NewCol("state", domain.UserSessionStateTerminated),
+			},
+			[]handler.Condition{
+				handler.NewCond("instance_id", event.Aggregate().InstanceID),
+				handler.NewCond("user_id", event.Aggregate().ID),
+			},
```

Comment:
Shouldn't we also add `state` as an condition, so we can limit the amount of updated rows? Else the update set will grow without bounds in the future and it won't scale well.


```suggestion
			[]handler.Condition{
				handler.NewCond("instance_id", event.Aggregate().InstanceID),
				handler.NewCond("user_id", event.Aggregate().ID),
				handler.NewCond("state", domain.UserSessionStateActive),
			},
```

Same proposal applies to the other event types.